src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -12,10 +12,23 @@ export function SwapSettingsSlippageInput({
   className,
   defaultSlippage = 3,
 }: SwapSettingsSlippageInputReact) {
-  const { setLifeCycleStatus } = useSwapContext();
-  const [slippage, setSlippage] = useState(defaultSlippage);
+  const { setLifeCycleStatus, lifeCycleStatus } = useSwapContext();
+  const getMaxSlippage = () => {
+    if (
+      !!lifeCycleStatus.statusData &&
+      'maxSlippage' in lifeCycleStatus.statusData
+    ) {
+      return lifeCycleStatus.statusData.maxSlippage;
```

Comment:
Two points here:
1. why are we checking if `statusData` is not there? should always be there.
2. we are making the logic more complex because `maxSlippage` is not in all state, so let's have it in all state, so the overl logic is simpler.

Reply:
This is overly complicated because I have to have these checks otherwise I get errors:

This is all to avoid Typescript errors. For example, if I just have:

` const maxSlippage = String(lifeCycleStatus.statusData.maxSlippage)`

Then I'll see the following errors: 

```
-  'lifeCycleStatus.statusData' is possibly 'null'.ts(18047)

-  'lifeCycleStatus.statusData' is possibly 'null'.ts(18047)

-  Property 'maxSlippage' does not exist on type 'SwapError | { isMissingRequiredField: boolean; } | { amountFrom: string; amountTo: string; tokenFrom?: Token | undefined; tokenTo?: Token | undefined; isMissingRequiredField: boolean; } | { ...; } | { ...; } | { ...; }'.Property 'maxSlippage' does not exist on type 'SwapError'.ts(2339)
```

Reply:
Two possibilities for adding maxSlippage in all states:

Add in every lifecycle statusData:
`    const maxSlippage = lifeCycleStatus.statusData.maxSlippage`
```
export type LifeCycleStatus =
  | {
      statusName: 'init';
      statusData: {
        isMissingRequiredField: boolean;
        maxSlippage: number;
      };
    }
  | {
      statusName: 'error';
      statusData: {
        maxSlippage: number;
        SwapError: SwapError,
      } 
    }
  | {
      statusName: 'amountChange';
      statusData: {
        amountFrom: string;
        amountTo: string;
        maxSlippage: number;
        tokenFrom?: Token;
        tokenTo?: Token;
        isMissingRequiredField: boolean;
      };
    }
  | {
      statusName: 'slippageChange';
      statusData: {
        maxSlippage: number;
      };
    }
  | {
      statusName: 'transactionPending';
      statusData: {
        maxSlippage: number;
      };
    }
  | {
      statusName: 'transactionApproved';
      statusData: {
        maxSlippage: number;
        transactionHash: Hex;
        transactionType: 'ERC20' | 'Permit2';
      };
    }
  | {
      statusName: 'success';
      statusData: {
        maxSlippage: number;
        transactionReceipt: TransactionReceipt;
      };
    };
```

Reply:
option 2:

export type LifeCycleStatus =
  | {
      statusName: 'init';
      statusData: {
        isMissingRequiredField: boolean;
      };
      maxSlippage: number;
    }
  | {
      statusName: 'error';
      statusData: SwapError;
      maxSlippage: number;
    }
  | {
      statusName: 'amountChange';
      statusData: {
        amountFrom: string;
        amountTo: string;
        tokenFrom?: Token;
        tokenTo?: Token;
        isMissingRequiredField: boolean;
      };
      maxSlippage: number;
    }
  | {
      statusName: 'transactionPending';
      statusData: null;
      maxSlippage: number;
    }
  | {
      statusName: 'transactionApproved';
      statusData: {
        transactionHash: Hex;
        transactionType: 'ERC20' | 'Permit2';
      };
      maxSlippage: number;
    }
  | {
      statusName: 'success';
      statusData: {
        transactionReceipt: TransactionReceipt;
      };
      maxSlippage: number;
    };

Reply:
Add everywhere, but not in the Error state.

Reply:
Nice, still need to check for the error state but much cleaner:

```
  const getMaxSlippage = () => {
    if (lifeCycleStatus.statusName !== 'error') {
      return lifeCycleStatus.statusData.maxSlippage;
    }
    return defaultSlippage;
  };
```

Reply:
dope